src/Adapter/AdapterAbstract.php

```diff
@@ -52,12 +54,8 @@
      */
     public function ok($output)
     {
-        $lines = explode("\n", $output);
-        foreach ($lines as $line) {
-            if (preg_match("%not ok \\d+ - %", $line)
-            && !preg_match("%# TODO%", $line)) {
-                return false;
-            }
+        if (preg_match("%##teamcity\[testFailed%", $output)) {
```

Comment:
Here's the short version that would work (excl. errors which return FALSE):
```php
return 0 === preg_match("%##teamcity\[testFailed%", $output);
```
Added via commit just now.